internal/storage/lru_cache.go

```diff
@@ -0,0 +1,153 @@
+// Licensed to the LF AI & Data foundation under one
```

Comment:
what is the difference between our cache and  https://github.com/hashicorp/golang-lru/blob/80c98217689d6df152309d574ccc682b21dc802c/simplelru/lru.go?